modules/_labs/button/react/lib/Button.tsx

```diff
@@ -0,0 +1,206 @@
+import * as React from 'react';
+import {colors} from '@workday/canvas-kit-react-core';
+import {focusRing, GrowthBehavior} from '@workday/canvas-kit-react-common';
+import {CanvasSystemIcon} from '@workday/design-assets-types';
+import {ButtonVariant, ButtonSize, ButtonColors, DropdownButtonVariant} from './types';
+import {ButtonContainer, ButtonLabel, ButtonLabelData, ButtonLabelIcon} from './parts';
+
+export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, GrowthBehavior {
+  /**
+   * The variant of the Button.
+   * @default ButtonVariant.Secondary
+   */
+  variant: ButtonVariant;
+  /**
+   * The size of the Button.
+   * @default ButtonSize.Medium
+   */
+  size: ButtonSize;
+  /**
+   * The ref to the button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * The data label of the Button.
+   */
+  dataLabel?: String;
+  /**
+   * The icon of the Button.
+   */
+  icon?: CanvasSystemIcon;
+}
+
+const Button = (props: ButtonProps) => {
+  const {variant, size, buttonRef, dataLabel, icon, children, ...elemProps} = props;
+
+  return (
+    // eslint-disable-next-line no-use-before-define
```

Comment:
We should either disable `no-use-before-define` for [functions](https://eslint.org/docs/rules/no-use-before-define#functions) in the linter or move the `getButtonColors` function prior to the class declaration.

I think this rule is generally useful to prevent "temporal dead zone" mistakes but in this case this comment seems like a lot of maintenance just to make sure our code is organized correctly. Oh JavaScript...